### ISSUE TITLE
Proposed extension encoding

### ISSUE THREAD

**Issue by 6677028** [bailey27/cppcryptfs#185]
I am using this software.
  I have another idea that you can add the feature to encode the file extensions.
Currently, the app has the option to encrypt file names. But with file name encryption, if that person deletes a character in the encrypted file name, the file will be corrupted.
  Therefore, to be safe, I think we just need to encode the file name extension into a similar extension name, such as .cpp.
Hope you will develop this app for a long time.

**Comment by yfdyh000** [comment]
I don't understand the meaning of this request.
Do you need optional encryption for the real extension? But the file cannot be opened.
Adding an extension does not resolve the corruption if the name is changed manually.
Alternatively, it is not secure to encrypt only the extension. Saving encrypted things to extension is not appropriate because they are too long.

**Comment by 6677028** [comment]
Thank you for your very quick reply.

#.   I use your app to encrypt my data. But the **File names** I chose are **Plain text**. I have some reason for not encrypting the filename.

           

- [ ]    1- I often have to copy and transfer files so I want to keep the file names. That way I can select the file that needs to be moved
- [ ]     2- when encrypting file names, if someone changes the encrypted file name then the file cannot be opened again.

#. As I know, Ransomware is usually attacked on certain file types like . doc, xlsx, PDF....so if you don't use encrypted file names, We will be attacked by Ransomware. Some encryption software also works well to change the extension like **Kruptos 2** which added the extension to **.K2p** or **Boxcryptor** which added the extension **.Bc** ....... So we don't need to encode the file name. so I want you to add such feature to CPPcryptfs. In my opinion you can add an extension to all encrypted files like **_.cpp_** and hide the default extension. Doing so will not affect filename encoding.

This is the way of Kruptos 2
<a href="https://imgur.com/okHC68v"><img src="https://i.imgur.com/okHC68v.png" title="source: imgur.com" /></a>
This is the way of Boxcryptor
<a href="https://imgur.com/VfCFIPy"><img src="https://i.imgur.com/VfCFIPy.png" title="source: imgur.com" /></a>

**Comment by 6677028** [comment]
Who can help me?

**Comment by red0888** [comment]
I agree, this is a very important feature that must be added. The final file extension could also be encrypted and appended to the encrypted file name and then accordingly, can identify and decrypt to the correct file extension. 

I see no reason why this shouldn't be added...

**Comment by tew42** [comment]
While I don't think it's a critical feature (for a great piece of software, by the way, thanks - had to migrate from my prior solution and quite liking this so far!), I do also think it would be useful for "plain text" filenames to add a generic extension (e.g., .cfs) when encrypted - not only for the ransomware concerns raised earlier, but also to prevent mixing up encrypted files & folders with unencrypted ones in day-to-day use:
Scenario 1: I might accidentally select a "recent file" to attach to an email that is in fact the encrypted version and thus useless to the recipient
Scenario 2: I might mess up some copy/pasting of files, putting an unencrypted file into the encrypted folder structure, which won't stand out in any way until I try to open it

An important question here, though, I guess, is whether introducing such a file-extension feature would still preserve the compatibility with gocryptfs, or if really this is a feature request that should go "upstream"?

**Comment by 6677028** [comment]
I am still using this software that has encrypted my data. I've always been following this issue. Hopefully, DEV can consider adding this feature to make the software complete

**Comment by ccchan234** [comment]
> I am using this software. I have another idea that you can add the feature to encode the file extensions. Currently, the app has the option to encrypt file names. But with file name encryption, if that person deletes a character in the encrypted file name, the file will be corrupted. Therefore, to be safe, I think we just need to encode the file name extension into a similar extension name, such as .cpp. Hope you will develop this app for a long time.

why you say so?

for my understanding,

u just need the .conf file, the diriv, and then the encrypted file.

suppose your encrypted file is ajhsdfljkahsdf8asda etc,

u can rename it, then it wont appear in the unencrypted window;
after that you simply rename it back and it will become normal? thanks
